connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestServer.java

```diff
@@ -280,6 +298,22 @@ ServerConnector findConnector(String protocol) {
         return null;
     }
 
+    void registerRestExtensions(Herder herder, ResourceConfig resourceConfig) {
+        connectRestExtensions = herder.plugins().newConnectRestExtensions(
+            config.getList(WorkerConfig.REST_EXTENSION_CLASSES_CONFIG),
+            config);
+
+        ConnectRestExtensionContext connectRestExtensionContext =
+            new ConnectRestExtensionContextImpl(
+                new ConnectRestConfigurable(resourceConfig),
+                new ConnectClusterStateImpl(herder)
+            );
+        for (ConnectRestExtension connectRestExtension : connectRestExtensions) {
+            connectRestExtension.register(connectRestExtensionContext);
```

Comment:
should we pass in `connectRestExtensionContext` with the `newConnectRestExtensions(..)` method above? it will be cleaner to create and register the plugin in one place.

Reply:
@wicknicks the method is now generic enough to instantiate any plugin. Irrespective, I think registering resources belongs in RestServer.